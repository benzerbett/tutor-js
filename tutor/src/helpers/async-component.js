import { asyncComponent as asyncLoader } from 'react-async-component';
import ErrorComponent from '../components/error-monitoring/async-load-error';
import LoadingComponent from '../components/loading-screen';

export function asyncComponent(resolve) {
  return asyncLoader({
    resolve,
    LoadingComponent,
    ErrorComponent,
  });
}

export function loadAsync(resolve) {
  // return a function so the router will only evaluate it when it's needed
  // in the future we can insert role dependant logic here
  return () => asyncComponent(resolve);
}
