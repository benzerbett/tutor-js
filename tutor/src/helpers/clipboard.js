import _ from 'underscore';

const SUPPORTED = typeof document.queryCommandSupported === 'function' ? document.queryCommandSupported('copy') : undefined;

export default {
  isSupported: _.constant(!!SUPPORTED),

  copy() {
    if (!SUPPORTED) { return; }

    try {
      return document.execCommand('copy');
    } catch (e) {
      return console.warn('clipboard copy failed', e);
    }
  },
};
