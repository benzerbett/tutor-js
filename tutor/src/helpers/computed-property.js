import { forIn } from 'lodash';
import { computed } from 'mobx';

export function addComputedProperty(obj, name, fn) {

  Object.defineProperty(obj, name, {
    configurable: true,
    get() {
      const getter = computed(fn, { context: this });
      Object.defineProperty(this, name, {
        configurable: false,
        get: () => getter.get(),
      });
      return getter.get();
    },
  });

}

export function extendHasMany(properties) {
  return (collection) => {
    forIn(properties, (fn, prop) => {
      addComputedProperty(collection, prop, fn);
    });
  };
}
