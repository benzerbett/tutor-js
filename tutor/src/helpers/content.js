import _ from 'underscore';


export default {
  // Convert each number to base 10 with it's position based on index.
  // If section is not present, 0 is set for it
  //   1 becomes 100, 1.1 becomes 101, 3.2 to 302, 3.2.1 -> 30201
  // Useful for sorting
  chapterSectionToNumber(chapter_section) {
    if (_.isString(chapter_section)) { chapter_section = chapter_section.split('.'); }
    if (chapter_section.length === 1) { chapter_section.push(0); } // add a section 0 if it has only a chapter
    let position = -1;
    return _.reduceRight(chapter_section, (memo, num) => memo + (num * Math.pow(100, (position += 1)))
      , 0);
  },
};
