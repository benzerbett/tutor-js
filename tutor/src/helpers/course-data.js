import Courses from '../models/courses-map';

import Router from '../helpers/router';
import TimeHelper from '../helpers/time';

export default {
  getCourseDataProps(courseId) {
    let dataProps;
    if (courseId == null) {
      ((((((({ courseId } = Router.currentParams())))))));
    }
    const course = Courses.get(courseId);
    return dataProps = {
      'data-title': course.name,
      'data-book-title': course.bookName || '',
      'data-appearance': course.appearance_code,
    };
  },

  getCourseBounds(courseId) {
    const course = Courses.get(courseId);

    const start = TimeHelper.getMomentPreserveDate(course.starts_at, TimeHelper.ISO_DATE_FORMAT);
    const end = TimeHelper.getMomentPreserveDate(course.ends_at, TimeHelper.ISO_DATE_FORMAT);

    return { start, end };
  },
};
