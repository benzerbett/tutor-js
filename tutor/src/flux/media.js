import _ from 'underscore';
import htmlparser from 'htmlparser2';
import { makeSimpleStore } from './helpers';

const LINKS_BEGIN = ['#'];
const LINKS_CONTAIN = ['cnx.org/contents/'];

const MEDIA_LINK_EXCLUDES = [
  '.nav',
  '.view-reference-guide',
  '[data-type=footnote-number]',
  '[data-type=footnote-ref]',
  '[data-targeted=media]',
  '[href*=\'/\']',
  '[href=\'#\']',
];

const buildAllowed = function(linksBegin, linksContain) {
  const beginSelectors = _.map(linksBegin, linkString => `a[href^='${linkString}']`);

  const containSelectors = _.map(linksContain, linkString => `a[href*='${linkString}']`);

  return _.union(beginSelectors, containSelectors);
};

const MediaConfig = {
  _local: {},

  loaded(id, mediaDOM) {
    this._local[id] = mediaDOM;
    return this.emit(`loaded.${id}`, mediaDOM);
  },

  _parseAndLoad(actions, dom, link) {
    if (link.attribs.href.search('#') === 0) {
      const id = link.attribs.href.replace('#', '');
      const idDOM = htmlparser.DomUtils.getElementById(id, dom);
      if (idDOM) {
        const idHTML = htmlparser.DomUtils.getOuterHTML(idDOM);

        const mediaDOM = {
          name: idDOM.name,
          html: idHTML,
        };

        return actions.loaded(id, mediaDOM);
      }
    }
  },

  _parseHandler(actions, error, dom) {
    const links = htmlparser.DomUtils.getElementsByTagName('a', dom);
    return _.each(links, _.partial(actions._parseAndLoad, actions, dom));
  },

  parse(htmlString) {
    if (this.parseHandler == null) { this.parseHandler = new htmlparser.DomHandler(_.partial(this._parseHandler, this)); }
    if (this.parser == null) { this.parser = new htmlparser.Parser(this.parseHandler); }
    return this.parser.parseComplete(htmlString);
  },

  _get(id) {
    return this._local[id];
  },

  reset() {
    this._local = {};
    delete this.parseHandler;
    return delete this.parser;
  },

  exports: {
    get(id) { return this._get(id); },
    isLoaded(id) { return (this._get(id) != null); },

    getMediaIds() {
      return _.keys(this._local);
    },

    getLinksContained() {
      return LINKS_CONTAIN;
    },

    getAllowed() {
      return buildAllowed(LINKS_BEGIN, LINKS_CONTAIN);
    },

    getExcluded() {
      return MEDIA_LINK_EXCLUDES;
    },

    getSelector() {
      const notMedias = _.reduce(MEDIA_LINK_EXCLUDES, (current, exclude) => `${current}:not(${exclude})`
        , '');

      return _.map(buildAllowed(LINKS_BEGIN, LINKS_CONTAIN), allowed => `${allowed}${notMedias}`).join(', ');
    },
  },
};


const { actions, store } = makeSimpleStore(MediaConfig);
export { actions as MediaActions, store as MediaStore };
