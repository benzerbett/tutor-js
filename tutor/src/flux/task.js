// coffeelint: disable=no_empty_functions
import {
  map, find, filter, each, values, findIndex, isEqual, first, pick,
  findLastIndex, isEmpty, pluck, uniq, compact,
} from 'lodash';
import moment from 'moment';
import flux from 'flux-react';

import Durations from '../helpers/durations';
import { CrudConfig, makeSimpleStore, extendConfig } from './helpers';
import { TaskStepActions, TaskStepStore } from './task-step';

import { MediaActions } from './media';
import { StepTitleActions } from './step-title';

const getSteps = steps =>
  map(steps, ({ id }) => TaskStepStore.get(id))
;

const getCurrentStepIndex = function(steps) {
  let currentStep = -1;
  for (let i = 0; i < steps.length; i++) {
    const step = steps[i];
    if (!step.is_completed) {
      currentStep = i;
      break;
    }
  }
  return currentStep;
};

const getCurrentStep = steps =>
  find(steps, step =>
    // return for first step where step.is_completed = false or
    // step.is_completed is undefined
    !step.is_completed || (step.is_completed == null)
  )
;

const getIncompleteSteps = steps =>
  filter(steps, step => (step != null) && !step.is_completed)
;

const getCompleteSteps = steps =>
  filter(steps, step => (step != null) && step.is_completed)
;

const getChangedSteps = steps =>
  filter(steps, step => (step != null) && TaskStepStore.isChanged(step.id))
;

// HACK When working locally a step completion triggers a reload but the is_completed field on the TaskStep
// is discarded. so, if is_completed is set on the local object but not on the returned JSON
// Tack on a dummy correct_answer_id
const hackLocalStepCompletion = function(step) {
  if (step.is_completed && (__guard__(__guard__(step.content != null ? step.content.questions : undefined, x1 => x1[0]), x => x.answers[0]) != null) && !step.correct_answer_id) {
    step.correct_answer_id = step.content.questions[0].answers[0].id;
    return step.feedback_html = 'Some <em>FAKE</em> feedback';
  }
};

const TaskConfig = {
  _steps: {},

  _getStep(taskId, stepId) {
    const step = find(this._steps[taskId], s => s.id === stepId);
    return step;
  },

  _grabHtmlFromReading(step) {
    return step.content_html;
  },

  _grabHtmlFromExercise(step) {
    if (step.content == null) { return ''; }

    let html = step.content.stimulus_html;
    const questionHtml = pluck(step.content.questions, 'stem_html').join('');
    return html += questionHtml;
  },

  _grabHtml(obj) {
    let htmlToParse;
    return htmlToParse = map(obj.steps, step => {
      let html;
      if (step.type === 'reading') {
        html = this._grabHtmlFromReading(step);
      } else if (step.type === 'exercise') {
        html = this._grabHtmlFromExercise(step);
      }
      return html;
    }).join('');
  },

  _loaded(obj, id) {
    this.emit('loaded', id);
    MediaActions.parse(this._grabHtml(obj));
    StepTitleActions.parseSteps(obj.steps);
    // Populate all the TaskSteps when a Task is loaded
    if (this._steps == null) { this._steps = {}; }
    // Remove the steps so Components are forced to use `.getSteps()` to get
    // the updated step objects
    const { steps } = obj;
    delete obj.steps;
    this._steps[id] = steps;

    for (let step of steps) {
      if (obj.HACK_LOCAL_STEP_COMPLETION) { hackLocalStepCompletion(step); }
      //HACK: set the task_id so we have a link back to the task from the step
      step.task_id = id;
      TaskStepActions.loaded(step, step.id);
    }
    obj;

    // explicit return obj to load onto @_local
    return obj;
  },

  completeStep(id, taskId) {
    TaskStepActions.complete(id);
    this.emit('step.completing', id);
    if (this.exports.hasPlaceholders.call(this, taskId) &&
      !this.exports.hasIncompleteCoreStepsIndexes.call(this, taskId)) {
      const placeholderSteps = this.exports.getPlaceholders.call(this, taskId);
      return each(placeholderSteps, step => TaskStepActions._loadPersonalized(step.id));
    }
  },

  stepCompleted(obj, taskStepId) {
    TaskStepActions.completed(obj, taskStepId);
    this.loaded(obj, obj.id);
    return this.emit('step.completed', taskStepId, obj.id);
  },

  exports: {
    getSteps(id) {
      if (!this._steps[id]) { throw new Error('BUG: Steps not loaded'); }
      return getSteps(this._steps[id]);
    },

    getAll() { return values(this._local); },

    getCurrentStepIndex(taskId) {
      const steps = getSteps(this._steps[taskId]);
      // Determine the first uncompleted step
      return getCurrentStepIndex(steps);
    },

    // Returns the reading and it's step index for a given task's ID
    getReadingForTaskId(taskId, id) {
      const steps = getSteps(this._steps[taskId]);
      const { related_content } = TaskStepStore.get(id);

      // replace findIndex with findLastIndex if we should be going to the
      // most recent step of a related reading
      const relatedStepIndex = findIndex(steps, step => (step.type === 'reading') && (isEqual(step.chapter_section, first(related_content).chapter_section)));

      // should never happen if the taskId was valid
      if (relatedStepIndex <= -1) { throw new Error('BUG: Invalid taskId.  Unable to find index'); }
      // Find the first step of the related reading that appears before the given task
      return { reading: steps[relatedStepIndex], index: relatedStepIndex };
    },

    getDefaultStepIndex(taskId) {
      const steps = getSteps(this._steps[taskId]);
      return getCurrentStepIndex(steps);
    },

    getStepsIds(id) {
      return map(this._steps[id], step => pick(step, 'id'));
    },

    getCurrentStep(taskId) {
      let step;
      const steps = getSteps(this._steps[taskId]);
      return step = getCurrentStep(steps);
    },

    getIncompleteSteps(taskId) {
      let steps;
      const allSteps = getSteps(this._steps[taskId]);
      return steps = getIncompleteSteps(allSteps);
    },

    getCompletedSteps(taskId) {
      let steps;
      const allSteps = getSteps(this._steps[taskId]);
      return steps = getCompleteSteps(allSteps);
    },

    getIncompleteCoreStepsIndexes(taskId) {
      const allSteps = getSteps(this._steps[taskId]);
      const firstIndex =   findIndex(allSteps, step => (step != null) && !step.is_completed && TaskStepStore.isCore(step.id));

      const lastIndex = findLastIndex(allSteps, step => (step != null) && !step.is_completed && TaskStepStore.isCore(step.id));

      const coreSteps = [
        firstIndex,
      ];
      if (lastIndex !== firstIndex) {
        coreSteps.push(lastIndex);
      }

      return coreSteps;
    },

    hasAnyStepChanged(taskId) {
      const allSteps = getSteps(this._steps[taskId]);
      return getChangedSteps(allSteps).length;
    },

    hasIncompleteCoreStepsIndexes(taskId) {
      const allSteps = getSteps(this._steps[taskId]);
      const steps = find(allSteps, step => (step != null) && !step.is_completed && TaskStepStore.isCore(step.id));
      return (steps != null);
    },

    getPlaceholders(taskId) {
      const allSteps = getSteps(this._steps[taskId]);
      return filter(allSteps, { type: 'placeholder' });
    },

    getFirstNonCoreIndex(taskId) {
      let stepIndex;
      const allSteps = getSteps(this._steps[taskId]);
      return stepIndex = findIndex(allSteps, step => (step != null) && !TaskStepStore.isCore(step.id));
    },

    hasPlaceholders(taskId) {
      return !isEmpty(this.exports.getPlaceholders.call(this, taskId));
    },

    isTaskCompleted(taskId) {
      const incompleteStep = getCurrentStep(getSteps(this._steps[taskId]));
      return !incompleteStep;
    },

    hasCrumbs(taskId) {
      return !(
        ((this._steps[taskId].length === 1) &&
        (this._get(taskId).type === 'external')) ||
        (this._get(taskId).type === 'reading')
      );
    },

    hasProgress(taskId) {
      return (this._steps[taskId].length >= 1) && (this._get(taskId).type === 'reading');
    },

    getRelatedSections(taskId) {
      return uniq(
        compact(
          pluck(
            getSteps(this._steps[taskId]), 'chapter_section'
          )
        )
      );
    },

    getStepsRelatedContent(taskId) {
      return sortBy(
        uniqBy(
          flatten(
            compact(
              pluck(
                filter(
                  getSteps(this._steps[taskId]),
                  step => TaskStepStore.isCore(step.id)
                ),
                'related_content'
              ),
            )
          ),
          cs => cs.chapter_section.join('.')
        ),
        cs => cs.chapter_section.join('.')
      );
    },

    getDetails(taskId) {
      let type;
      let title = '';
      let sections = [];

      ({ title, type } = this._get(taskId));
      sections = this.exports.getRelatedSections.call(this, taskId);

      if (isEmpty(sections) && (type === 'concept_coach')) {
        const details = this.exports.getStepsRelatedContent.call(this, taskId);
        if (!isEmpty(details)) {
          sections = pluck(details, 'chapter_section');
          ({ title } = details[0]);
        }
      }

      return { title, sections };
    },

    getCompletedStepsCount(taskId) {
      const allSteps = getSteps(this._steps[taskId]);
      const steps = getCompleteSteps(allSteps);

      return steps.length;
    },

    getTotalStepsCount(taskId) {
      const allSteps = getSteps(this._steps[taskId]);

      return allSteps.length;
    },

    isTaskPastDue(taskId) {
      return Durations.isPastDue( this._get(taskId) );
    },

    isPractice(taskId) {
      const practices = [
        'practice',
        'chapter_practice',
        'page_practice',
        'practice_worst_topics',
      ];

      if (practices.indexOf(this._get(taskId).type) > -1) { return true; } else { return false; }
    },

    getStepIndex(taskId, stepId) {
      return findIndex(this._steps[taskId], { id: stepId });
    },

    getStepLateness(taskId, stepId) {
      const result = {
        late: false,
        last_completed_at: null,
        how_late: null,
      };

      const step = this._getStep(taskId, stepId);
      const { due_at, type } = this._get(taskId);

      if ((step == null) || (type !== 'homework')) { return result; }

      const { last_completed_at } = step;

      result.late = moment(due_at).isBefore(last_completed_at);
      result.last_completed_at = last_completed_at;
      result.how_late = moment(due_at).from(last_completed_at, true);

      return result;
    },

    getStepParts(taskId, stepId) {
      const currentStep = this._getStep(taskId, stepId);
      const { content_url } = currentStep;
      let parts = filter(this._steps[taskId], step => step.is_in_multipart && (step.content_url === content_url));

      if (isEmpty(parts)) { parts = [currentStep]; }

      return parts = getSteps(parts);
    },

    getStepByIndex(taskId, stepIndex) {
      return this._steps[taskId][stepIndex];
    },

    isFeedbackImmediate(taskId) {
      const { is_feedback_available } = this._get(taskId);
      return is_feedback_available;
    },

    isDeleted(taskId) {
      const { is_deleted } = this._get(taskId);
      return is_deleted;
    },

    isSameStep(taskId, ...stepIds) {
      const contentUrls = uniq(
        map(stepIds, stepId => {
          const step = this._getStep(taskId, stepId);

          if ((step != null ? step.is_in_multipart : undefined)) {
            return step.content_url;
          } else {
            return null;
          }
        })
      );
      return (contentUrls.length === 1) && (first(contentUrls) != null);
    },
  },
};

extendConfig(TaskConfig, new CrudConfig());
const { actions, store } = makeSimpleStore(TaskConfig);
export { actions as TaskActions, store as TaskStore };

function __guard__(value, transform) {
  return (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
}
