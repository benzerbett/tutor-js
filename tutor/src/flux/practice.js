import _ from 'underscore';
import hash from 'object-hash';

import { TaskActions, TaskStore } from './task';
import { CrudConfig, makeSimpleStore, extendConfig, STATES } from './helpers';

const getTopicHash = _.partial(hash, _, { unorderedArrays: true });
const makePractice = (
  result,
  courseId,
  topicParams,
) => ({
  id: result.id,
  created_for: topicParams,
  exists: true,
})
;

const CoursePractice = {
  _topics: {},
  _cache: {},

  _reset() {
    this._topics = {};
    return this._cache = {};
  },

  _recordTopics(taskId, courseId, topicParams) {
    if (this._topics[courseId] == null) { this._topics[courseId] = {}; }
    return this._topics[courseId][taskId] = topicParams;
  },

  _addToCache(thingToCache, courseId, topicParams) {
    const topicHash = getTopicHash(topicParams);
    if (_.isEmpty(topicHash)) { return; }

    if (this._cache[courseId] == null) { this._cache[courseId] = {}; }
    if (this._cache[courseId][topicHash] == null) { this._cache[courseId][topicHash] = []; }
    return this._cache[courseId][topicHash].push(thingToCache);
  },

  _cacheError(result, courseId, topicParams) {
    return this._addToCache({ exists: false }, courseId, topicParams);
  },

  _cacheSuccess(result, courseId, topicParams) {
    const practice = makePractice(result, courseId, topicParams);
    this._addToCache(practice, courseId, topicParams);
    return practice;
  },

  _failed(result, { courseId, query }) {
    return this._cacheError(result, courseId, query);
  },

  create(courseId, topicParams) {
    if (!this.dontReload(courseId)) { this._local[courseId] = {}; }
    return this._asyncStatus[courseId] = STATES.LOADED;
  },

  created(result, courseId, topicParams) {
    // this will use the base config's loaded, which will
    // run _loaded within.
    return this.loaded(result, courseId, topicParams);
  },

  _loaded(result, { courseId, query }) {
    TaskActions.loaded(result, result.id);

    this._recordTopics(result.id, courseId, query);
    // cache and return practice for loading onto _local
    this._cacheSuccess(result, courseId, query);
    return this.emit(`${STATES.LOADED}.${courseId}`, result.id);
  },

  _getFromCache(courseId, topicParams) {
    const topicHash = getTopicHash(topicParams);
    if (_.isEmpty(topicHash)) { return; }

    return _.last(this._cache[courseId] != null ? this._cache[courseId][topicHash] : undefined);
  },

  _get(courseId, topicParams) {
    if (_.isEmpty(topicParams)) {
      return this._local[courseId];
    } else {
      return this._getFromCache(courseId, topicParams);
    }
  },

  exports: {
    getTaskId(courseId, topicParams) {
      return __guard__(this._get(courseId, topicParams), x => x.id);
    },

    getPageIds(courseId, topicParams) {
      return __guard__(__guard__(this._get(courseId, topicParams), x1 => x1.created_for), x => x.page_ids);
    },

    has(courseId, topicParams) {
      return __guard__(this._get(courseId, topicParams), x => x.exists);
    },

    get(courseId, topicParams) {
      if (!this.exports.has.call(this, courseId, topicParams)) { return {}; }

      const taskId = this.exports.getTaskId.call(this, courseId, topicParams);
      return TaskStore.get(taskId);
    },

    isDisabled(courseId, topicParams) {
      return (this._get(courseId, topicParams) != null) && (this._get(courseId, topicParams).exists === false);
    },

    getCurrentTopics(courseId, taskId) {
      return (this._topics[courseId] != null ? this._topics[courseId][taskId] : undefined) || {};
    },
  },
};

extendConfig(CoursePractice, new CrudConfig());
const { actions, store } = makeSimpleStore(CoursePractice);
export { actions as CoursePracticeActions, store as CoursePracticeStore };

function __guard__(value, transform) {
  return (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
}