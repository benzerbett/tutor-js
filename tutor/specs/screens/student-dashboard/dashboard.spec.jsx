import { React, SnapShot, Wrapper } from '../../helpers';
import Dashboard from '../../../src/screens/student-dashboard/dashboard';
import Factory from '../../factories';
import { bootstrapCoursesList } from '../../courses-test-data';
import chronokinesis from 'chronokinesis';
import moment from 'moment-timezone';
import { TimeActions } from '../../../src/flux/time';

describe('Student Dashboard', () => {
  let props;

  beforeEach(() => {
    moment.tz.setDefault('America/Chicago');
    moment.locale('en');
    const now = new Date('2015-10-14T12:00:00.000Z');
    chronokinesis.travel(now);
    const course = Factory.course();
    bootstrapCoursesList();
    Factory.studentTasks({ course, attributes: { now: new Date('2015-10-21T12:00:00.000Z') } });
    props = {
      course,
      params: {},
    };
  });

  afterEach(() => {
    chronokinesis.reset();
    moment.tz.setDefault();
  });

  it('matches snapshot', function() {
    props.course.studentTasks.all_tasks_are_ready = false;
    props.course.primaryRole.joined_at = new Date('2015-09-14T12:00:00.000Z');
    const component = SnapShot.create(
      <Wrapper _wrapped_component={Dashboard} {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays as loading', () => {
    props.course.studentTasks.all_tasks_are_ready = false;
    props.course.primaryRole.joined_at = new Date('2015-10-14T12:00:00.000Z');
    const component = SnapShot.create(
      <Wrapper _wrapped_component={Dashboard} {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
