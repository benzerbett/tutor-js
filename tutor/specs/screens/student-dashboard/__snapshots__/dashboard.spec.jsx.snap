// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Dashboard displays as loading 1`] = `
<div
  className="student-dashboard"
>
  <div
    className="openstax-notifications-bar"
  />
  <div
    className="course-title-banner"
    data-appearance="testing"
    data-book-title="Testing"
    data-is-preview={false}
    data-term="Spring 2017"
    data-title="Local Test Course Two"
  >
    <div
      className="book-title"
    >
      <h1
        className="book-title-text"
      >
        Local Test Course Two
      </h1>
    </div>
    <div
      className="course-term"
    >
      Spring 2017
    </div>
  </div>
  <div
    className="container"
    data-tour-region-id="student-dashboard"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-9 col-md-8 col-xs-12"
      >
        <nav
          className="tutor-tabs"
        >
          <ul
            className="nav nav-tabs"
            role="tablist"
          >
            <li
              aria-selected={true}
              className="active"
              role="tab"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <h2>
                  This Week
                </h2>
              </a>
            </li>
            <li
              aria-selected={false}
              className=""
              role="tab"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <h2>
                  All Past Work
                </h2>
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="tab-pane active"
          role="tabpanel"
        >
          <div
            className="empty pending panel panel-default"
            id={undefined}
          >
            <div
              className="panel-body"
            >
              <i
                className="tutor-icon fa fa-spinner fa-spin"
                role="presentation"
                type="spinner"
              />
               Preparing assignments for your course.  This can take up to 10 minutes.
            </div>
          </div>
          <div
            className="empty pending upcoming panel panel-default"
            id={undefined}
          >
            <div
              className="panel-body"
            >
              <i
                className="tutor-icon fa fa-spinner fa-spin"
                role="presentation"
                type="spinner"
              />
               Preparing assignments for your course.  This can take up to 10 minutes.
            </div>
          </div>
        </div>
      </div>
      <div
        className="sidebar col-lg-3 col-md-4 col-xs-12"
      >
        <div
          className="actions-box loadable is-loading"
        >
          Loading progress information... 
        </div>
        <div
          className="actions-box"
        >
          <a
            aria-label="Browse course reference book"
            className="browse-the-book"
            data-appearance="college_physics"
            href="/book/1"
            onClick={[Function]}
            target="_blank"
          >
            <div>
              Browse the Book
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Dashboard matches snapshot 1`] = `
<div
  className="student-dashboard"
>
  <div
    className="openstax-notifications-bar viewable"
  >
    <div
      className="notification missing-student-id"
    >
      <span
        className="body"
      >
        <i
          className="icon fa fa-info-circle"
        />
        <span>
          To get credit for your work, add your student ID.
        </span>
        <a
          className="action"
          onClick={[Function]}
        >
          Add Student ID
        </a>
      </span>
    </div>
  </div>
  <div
    className="course-title-banner"
    data-appearance="testing"
    data-book-title="Testing"
    data-is-preview={false}
    data-term="Spring 2017"
    data-title="Local Test Course One"
  >
    <div
      className="book-title"
    >
      <h1
        className="book-title-text"
      >
        Local Test Course One
      </h1>
    </div>
    <div
      className="course-term"
    >
      Spring 2017
    </div>
  </div>
  <div
    className="container"
    data-tour-region-id="student-dashboard"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-9 col-md-8 col-xs-12"
      >
        <nav
          className="tutor-tabs"
        >
          <ul
            className="nav nav-tabs"
            role="tablist"
          >
            <li
              aria-selected={true}
              className="active"
              role="tab"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <h2>
                  This Week
                </h2>
              </a>
            </li>
            <li
              aria-selected={false}
              className=""
              role="tab"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                <h2>
                  All Past Work
                </h2>
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="tab-pane active"
          role="tabpanel"
        >
          <div
            className="-this-week panel panel-default"
            id={undefined}
          >
            <div
              className="panel-body"
            >
              <div
                className="row labels"
              >
                <div
                  className="col-sm-7 col-xs-12"
                >
                  <span
                    className="date-range"
                  >
                    <time>
                      Oct 12, 2015
                    </time>
                    <span>
                      â€“
                    </span>
                    <time>
                      Oct 18, 2015
                    </time>
                  </span>
                </div>
                <div
                  className="progress-label col-sm-3 col-sm-offset-0 col-xs-5 col-xs-offset-2"
                >
                  Progress
                </div>
                <div
                  className="due-at-label col-sm-2 col-xs-5"
                >
                  Due
                </div>
              </div>
              <a
                aria-label="Work on reading: dot-com recontextualize communities"
                className="task row reading"
                data-event-id={1}
                href="/course/1/task/1"
                onClick={[Function]}
                onKeyDown={undefined}
                tabIndex={-1}
              >
                <div
                  className="column-icon col-sm-1 col-xs-2"
                >
                  <i
                    aria-label="reading icon"
                    className="icon icon-lg icon-reading"
                  />
                </div>
                <div
                  className="title col-sm-6 col-xs-10"
                >
                  dot-com recontextualize communities
                </div>
                <div
                  className="feedback col-sm-3 col-xs-5"
                >
                  <span>
                    In progress
                  </span>
                </div>
                <div
                  className="due-at col-sm-2 col-xs-5"
                >
                  <time>
                    Oct 18, 7:00am
                  </time>
                </div>
              </a>
              <a
                aria-label="Work on external: out-of-the-box embrace supply-chains"
                className="task row external"
                data-event-id={2}
                href="/course/1/task/2"
                onClick={[Function]}
                onKeyDown={undefined}
                tabIndex={-1}
              >
                <div
                  className="column-icon col-sm-1 col-xs-2"
                >
                  <i
                    aria-label="external icon"
                    className="icon icon-lg icon-external"
                  />
                </div>
                <div
                  className="title col-sm-6 col-xs-10"
                >
                  out-of-the-box embrace supply-chains
                </div>
                <div
                  className="feedback col-sm-3 col-xs-5"
                >
                  <span>
                    Not started
                  </span>
                </div>
                <div
                  className="due-at col-sm-2 col-xs-5"
                >
                  <time>
                    Oct 18, 7:00am
                  </time>
                </div>
              </a>
              <a
                aria-label="Work on event: B2B deliver content"
                className="task row event"
                data-event-id={3}
                href="/course/1/task/3"
                onClick={[Function]}
                onKeyDown={undefined}
                tabIndex={-1}
              >
                <div
                  className="column-icon col-sm-1 col-xs-2"
                >
                  <i
                    aria-label="event icon"
                    className="icon icon-lg icon-event"
                  />
                </div>
                <div
                  className="title col-sm-6 col-xs-10"
                >
                  B2B deliver content
                </div>
                <div
                  className="feedback col-sm-3 col-xs-5"
                >
                  <span>
                    
                  </span>
                </div>
                <div
                  className="due-at col-sm-2 col-xs-5"
                >
                  <time>
                    Oct 18, 7:00am
                  </time>
                </div>
              </a>
              <a
                aria-label="Work on homework: virtual strategize infomediaries"
                className="task row homework"
                data-event-id={4}
                href="/course/1/task/4"
                onClick={[Function]}
                onKeyDown={undefined}
                tabIndex={-1}
              >
                <div
                  className="column-icon col-sm-1 col-xs-2"
                >
                  <i
                    aria-label="homework icon"
                    className="icon icon-lg icon-homework"
                  />
                </div>
                <div
                  className="title col-sm-6 col-xs-10"
                >
                  virtual strategize infomediaries
                </div>
                <div
                  className="feedback col-sm-3 col-xs-5"
                >
                  <span>
                    <span
                      className="tour-anchor"
                      data-tour-anchor-id="assignment-progress-status"
                    >
                      10/11 answered
                    </span>
                  </span>
                </div>
                <div
                  className="due-at col-sm-2 col-xs-5"
                >
                  <time>
                    Oct 18, 7:00am
                  </time>
                </div>
              </a>
            </div>
          </div>
          <div
            className="empty upcoming panel panel-default"
            id={undefined}
          >
            <div
              className="panel-body"
            >
              No upcoming assignments
            </div>
          </div>
        </div>
      </div>
      <div
        className="sidebar col-lg-3 col-md-4 col-xs-12"
      >
        <div
          className="actions-box loadable is-loading"
        >
          Loading progress information... 
        </div>
        <div
          className="actions-box"
        >
          <a
            aria-label="Browse course reference book"
            className="browse-the-book"
            data-appearance="college_physics"
            href="/book/1"
            onClick={[Function]}
            target="_blank"
          >
            <div>
              Browse the Book
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
